lib/bson.js

```diff
@@ -176,212 +191,266 @@ BSON.prototype.calculateObjectSize = function(object, options) {
  **/
-BSON.BSON_DATA_INT = 16;
+const BSON_DATA_INT = 16;
+
 /**
  * Timestamp BSON Type
  *
  * @classconstant BSON_DATA_TIMESTAMP
  **/
-BSON.BSON_DATA_TIMESTAMP = 17;
+const BSON_DATA_TIMESTAMP = 17;
+
 /**
  * Long BSON Type
  *
  * @classconstant BSON_DATA_LONG
  **/
-BSON.BSON_DATA_LONG = 18;
+const BSON_DATA_LONG = 18;
+
 /**
  * MinKey BSON Type
  *
  * @classconstant BSON_DATA_MIN_KEY
  **/
-BSON.BSON_DATA_MIN_KEY = 0xff;
+const BSON_DATA_MIN_KEY = 0xff;
+
 /**
  * MaxKey BSON Type
  *
  * @classconstant BSON_DATA_MAX_KEY
  **/
-BSON.BSON_DATA_MAX_KEY = 0x7f;
+const BSON_DATA_MAX_KEY = 0x7f;
 
 /**
  * Binary Default Type
  *
  * @classconstant BSON_BINARY_SUBTYPE_DEFAULT
  **/
-BSON.BSON_BINARY_SUBTYPE_DEFAULT = 0;
+const BSON_BINARY_SUBTYPE_DEFAULT = 0;
+
 /**
  * Binary Function Type
  *
  * @classconstant BSON_BINARY_SUBTYPE_FUNCTION
  **/
-BSON.BSON_BINARY_SUBTYPE_FUNCTION = 1;
+const BSON_BINARY_SUBTYPE_FUNCTION = 1;
+
 /**
  * Binary Byte Array Type
  *
  * @classconstant BSON_BINARY_SUBTYPE_BYTE_ARRAY
  **/
-BSON.BSON_BINARY_SUBTYPE_BYTE_ARRAY = 2;
+const BSON_BINARY_SUBTYPE_BYTE_ARRAY = 2;
+
 /**
  * Binary UUID Type
  *
  * @classconstant BSON_BINARY_SUBTYPE_UUID
  **/
-BSON.BSON_BINARY_SUBTYPE_UUID = 3;
+const BSON_BINARY_SUBTYPE_UUID = 3;
+
 /**
  * Binary MD5 Type
  *
  * @classconstant BSON_BINARY_SUBTYPE_MD5
  **/
-BSON.BSON_BINARY_SUBTYPE_MD5 = 4;
+const BSON_BINARY_SUBTYPE_MD5 = 4;
+
 /**
  * Binary User Defined Type
  *
  * @classconstant BSON_BINARY_SUBTYPE_USER_DEFINED
  **/
-BSON.BSON_BINARY_SUBTYPE_USER_DEFINED = 128;
-
-// Return BSON
-module.exports = BSON;
-module.exports.Code = Code;
-module.exports.Map = Map;
-module.exports.Symbol = Symbol;
-module.exports.BSON = BSON;
-module.exports.DBRef = DBRef;
-module.exports.Binary = Binary;
-module.exports.ObjectID = ObjectID;
-module.exports.Long = Long;
-module.exports.Timestamp = Timestamp;
-module.exports.Double = Double;
-module.exports.Int32 = Int32;
-module.exports.MinKey = MinKey;
-module.exports.MaxKey = MaxKey;
-module.exports.BSONRegExp = BSONRegExp;
-module.exports.Decimal128 = Decimal128;
+const BSON_BINARY_SUBTYPE_USER_DEFINED = 128;
+
+module.exports = {
+  // constants
+  BSON_INT32_MAX,
+  BSON_INT32_MIN,
+  BSON_INT64_MAX,
+  BSON_INT64_MIN,
+  JS_INT_MAX,
+  JS_INT_MIN,
+  BSON_DATA_NUMBER,
+  BSON_DATA_STRING,
+  BSON_DATA_OBJECT,
+  BSON_DATA_ARRAY,
+  BSON_DATA_BINARY,
+  BSON_DATA_OID,
+  BSON_DATA_BOOLEAN,
+  BSON_DATA_DATE,
+  BSON_DATA_NULL,
+  BSON_DATA_REGEXP,
+  BSON_DATA_CODE,
+  BSON_DATA_SYMBOL,
+  BSON_DATA_CODE_W_SCOPE,
+  BSON_DATA_INT,
+  BSON_DATA_TIMESTAMP,
+  BSON_DATA_LONG,
+  BSON_DATA_MIN_KEY,
+  BSON_DATA_MAX_KEY,
+  BSON_BINARY_SUBTYPE_DEFAULT,
+  BSON_BINARY_SUBTYPE_FUNCTION,
+  BSON_BINARY_SUBTYPE_BYTE_ARRAY,
+  BSON_BINARY_SUBTYPE_UUID,
+  BSON_BINARY_SUBTYPE_MD5,
+  BSON_BINARY_SUBTYPE_USER_DEFINED,
+
+  // wrapped types
+  Code,
+  Map,
+  Symbol,
+  DBRef,
+  Binary,
+  ObjectID,
+  ObjectId: ObjectID,
```

Comment:
If we're going to bump the major version, maybe now's a good time to drop this alias?

Reply:
Yeah good idea

Reply:
I did the work to rename everything internally to use the preferred `ObjectId`, but I just can't quite bite the bullet dropping the alias. It's so cheap for us, and it has such potential to break everyone's code. If everyone feels very strongly that this is the right path I'd reconsider, but otherwise it feels like we're begging for bug reports.

Reply:
Add a deprecated getter to return it, so people get a deprecation message?